pkg/controllers/workloadrebalancer/workloadrebalancer_controller.go

```diff
@@ -84,18 +92,34 @@ func (c *RebalancerController) Reconcile(ctx context.Context, req controllerrunt
 	}
 
 	// 2. get and update referenced binding to trigger a rescheduling
-	newStatus, successNum, retryNum := c.doWorkloadRebalance(ctx, rebalancer)
+	result := c.doWorkloadRebalance(ctx, rebalancer)
 
-	// 3. update status of WorkloadRebalancer
-	if err := c.updateWorkloadRebalancerStatus(ctx, rebalancer, newStatus); err != nil {
-		return controllerruntime.Result{}, err
+	// 3. update status of WorkloadRebalancer if statusChanged
```

Comment:
Is step 3 part of step 2, or can the status update be put into method `doWorkloadRebalance`?

Reply:
I don't think step 3 is part of step 2, these 3 steps are parallel: read rebalancer -> work according to rebalancer -> write rebalancer

Reply:
done